scripts/concat_seq_results.py

```diff
@@ -0,0 +1,291 @@
+
+import argparse
+import logging
+import re
+import sys
+from datetime import date
+
+import pandas as pd
+
+__author__ = "CDPHE"
+__copyright__ = "State of Colorado"
+__license__ = "GPL-3.0.0-or-later"
+
+log = logging.getLogger(__name__)
+
+
+def parse_args(args: list[str]) -> argparse.Namespace:
+    """Parse command line arguments."""
+    parser = argparse.ArgumentParser(description="Sequencing results summary.")
+
+    parser.add_argument(
+        "--log_level",
+        help="the level to log at",
+        choices=["CRITICAL", "ERROR", "WARNING", "INFO", "DEBUG"],
+        default="INFO",
+    )
+    parser.add_argument(
+        "--workflow_name",
+        help="the workflow name (i.e. what is on GitHub)",
+        default="CDPHE-Viral-Amplicon-WDL",
+    )
+    parser.add_argument(
+        "--workflow_version",
+        help="the workflow version (i.e. what is on GitHub)",
+        required=True,
+    )
+    parser.add_argument(
+        "--platform",
+        help="the sequencing platform",
+        choices=["illumina", "ont"],
+        default="illumina",
+    )
+    parser.add_argument("--sample_name_array")
+    parser.add_argument("--workbook_path")
+    parser.add_argument("--cov_out_files", help="txt file with list of bam file paths")
+    parser.add_argument(
+        "--percent_cvg_files", help="txt file with list of percent cvg file paths"
+    )
+    parser.add_argument(
+        "--nextclade_csv_files", help="txt file with list of nextclade csv file paths"
+    )
+    parser.add_argument("--assembler_version")
+    parser.add_argument("--project_name")
+
+    return parser.parse_args(args)
+
+
+def setup_logging(log_level: str) -> None:
+    """Set up logging."""
+    log_format = "[%(asctime)s] %(levelname)s:%(name)s:%(message)s"
+    logging.basicConfig(
+        level=log_level,
+        stream=sys.stdout,
+        format=log_format,
+        datefmt="%Y-%m-%d %H:%M:%S",
+    )
+
+
+def create_list_from_write_lines_input(write_lines_input: str) -> list[str]:
+    """Create list from write lines input."""
+    list = []
+    with open(write_lines_input, "r") as f:
+        for line in f:
+            list.append(line.strip())
+    return list
+
+
+def concat_cov_out(cov_out_file_list: list[str]) -> pd.DataFrame:
+    """Concatenate covergate output files."""
+    # initiate dataframe for concatenation
+    df = pd.DataFrame()
+    sample_name_list = []
+    samtools_mapped_reads_list = []
+    samtools_depth_list = []
+    samtools_baseq_list = []
+    samtools_mapq_list = []
+
+    # loop through bam file stats files and pull data
+    for file in cov_out_file_list:
+        d = pd.read_csv(file, sep="\t")
+        if re.search("barcode", file):
+            # for nanopore runs
+            sample_name = re.findall(r"/([0-9a-zA-Z_\-\.]+)_barcode", file)[0]
+        else:
+            # for illumina runs
+            sample_name = re.findall(r"/([0-9a-zA-Z_\-\.]+)_coverage.txt", file)[0]
```

Comment:
Workflows are Illumina-only, also the barcode naming may be CDPHE-specific

```suggestion
        sample_name = re.findall(r"/([0-9a-zA-Z_\-\.]+)_coverage.txt", file)[0]
```